hcl2template/types.packer_config.go

```diff
@@ -411,6 +510,17 @@ func (cfg *PackerConfig) GetBuilds(opts packer.GetBuildsOptions) ([]packer.Build
 	return res, diags
 }
 
+func setBuildVariables(generatedVars map[string]string) Variables {
+	variables := make(Variables)
+	for k := range generatedVars {
+		variables[k] = &Variable{
+			DefaultValue: cty.StringVal("unknown"),
```

Comment:
Using `cty.UnknownVal(cty.String)` fails the `Prepare` for both Provisioners and Post-Processors

Reply:
Got it, this can be worked on a bit later on I think 🙂

Reply:
Is there a problem with setting a default string as `unknown`?